test/llvm/llvm-invariant/function_local_const.chpl

```diff
@@ -0,0 +1,21 @@
+// This test checks if llvm.invariant.start is added to function local constants
+// Normally primitive type constants could be used, but chapel compiler
+// performs some constant propagation, which doesn't happen in case of records.
+
+record A
+{
+  var a : int;
+}
```

Comment:
@mppf 
I'm not sure what should I test as initializers make creating new records differently:

Here is how constructors work:
```
%tmp = alloca %A_chpl;
%localConst = alloca %A_chpl;
call construct_A(%A_chpl* %tmp, i64 %arg);
store %A_chpl %tmp, %A_chpl* %localConst
```

and here is how initializers work:

```
%localConst = alloca %A_chpl;
call init_A(%A_chpl* %tmp, i64 %arg);
```

Thus there is no final store which would mark that this variable is constant from this point giving opportunity to mark it using llvm.invariant.start, my algorithm covers a lot of cases because chapel compiler uses `store from temporary to final` a lot while in this case it doesn't happen so it doesn't do that.